pkg/webhooks/webhooks.go

```diff
@@ -220,21 +223,23 @@ func HealthProbe(ctx context.Context) healthz.Checker {
 	}
 }
 
-func ValidateConversionEnabled(ctx context.Context, kubeclient client.Client) {
-	// allow context to exist longer than cache sync timeout which has a default of 120 seconds
-	listCtx, cancel := context.WithTimeout(ctx, 130*time.Second)
-	defer cancel()
-	var err error
-	v1np := &v1.NodePoolList{}
+func ValidateConversionEnabled(ctx context.Context, directKubeClient client.Client) {
 	for {
-		err = kubeclient.List(listCtx, v1np, &client.ListOptions{Limit: 1})
-		if err == nil {
-			return
+		// We only need to try to get a single NodePool teh
+		err := directKubeClient.List(ctx, &v1.NodePoolList{}, &client.ListOptions{Limit: 1})
+		// We only want to bail out with a failure if we know the reason that we are not able to List on this API
+		// is due to an internal error on the conversion webhook
+		if errors.IsInternalError(err) && strings.Contains(err.Error(), "conversion webhook for karpenter.sh/v1beta1, Kind=NodePool failed") {
```

Comment:
Can we also not get an error as such:
`conversion webhook for karpenter.sh/v1, Kind=NodePool failed`

Reply:
No, since this is going to just be pulling the v1 version and there isn't going to be any conversion from v1 to v1

Reply:
I imagine we shouldn't see this if the stored version is v1 since we'll get the object directly without using the conversion webhook. I do wonder though, what happens when there's a mix of v1 and v1beta1 stored resources in the cluster? Since we're limiting the list to a single resource, could we get a false negative if the resource we get is v1?